### PR TITLE
Use Patch when syncing GCP BackendService. Linking backend groups still uses Update

### PR DESCRIPTION
**Pull request by gauravkghildiyal**
/assign @swetharepakula

### CODE REVIEW

File: pkg/backends/syncer.go [hunk 1]
```diff
@@ -305,6 +305,9 @@ func ensureProtocol(be *composite.BackendService, p utils.ServicePort) (needsUpd
 	if be.Protocol == string(p.Protocol) {
 		return false
 	}
+
+	fmt.Printf("protocol needs change; got be = %+v, want protocol = %v\n", be, p.Protocol)
```
Comment (swetharepakula) [inline pkg/backends/syncer.go:309]: Should this be a log line instead?

Comment (gauravkghildiyal) [reply pkg/backends/syncer.go]: Deleted. That log line shouldn't exist. I'm not really sure how it got there. I don't remember adding it.

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**